### PR TITLE
Add operational document on configuration scoping

### PR DESCRIPTION
**Pull request by howardjohn**
After linking https://karlstoney.com/istio-at-scale-sidecar/ 1000 times, probably a good idea to make it first class (thanks Karl!!)

### CODE REVIEW

File: content/en/docs/ops/deployment/performance-and-scalability/index.md [hunk 1]
```diff
@@ -54,13 +54,13 @@ The CPU consumption scales with the following factors:
 - The rate of configuration changes.
 - The number of proxies connecting to Istiod.
 
-however this part is inherently horizontally scalable.
+However, this part is inherently horizontally scalable.
 
-When [namespace isolation](/docs/reference/config/networking/sidecar/) is enabled,
```
Comment (howardjohn) [inline content/en/docs/ops/deployment/performance-and-scalability/index.md:59]: This change is not strictly related ,but these numbers are basically made up junk from Istio 1.0. Its misleading to users to claim such a simple mapping of services => CPU.

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+{{</ text >}}
+
+{{< warning >}}
+Istiod will always open a watch to Kubernetes for all namespaces.
+However, discovery selectors will ignore objects that are not selected very early in its processing, minimizing costs.
+{{</ warning >}}
+
+## Frequently asked questions
+
+### How can I understand the cost of a certain configuration
```
Comment (dhawton) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:98]:
```suggestion
### How can I understand the cost of a certain configuration?
```

**Review by dhawton — COMMENTED** [review]
One minor nit for consistency, otherwise LGTM

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: details
+  annotations:
+    networking.istio.io/exportTo: ".,client"
+spec: ...
+{{</ text >}}
```
Comment (frankbu) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:70]: I think this is the source of your lint errors

```suggestion
{{< /text >}}
```

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+description: Shows how to scope configuration in Istio, for operational and performance benefits.
+weight: 10
+keywords: [scalability]
+owner: istio/wg-networking-maintainers
+test: no
+---
+
+In order to program the service mesh, the Istio control plane (Istiod) reads a variety of configurations, including core Kubernetes types like `Service` and `Node`,
+and Istio's own types like `Gateway`.
+These are then sent the data plane (see [Architecture](/docs/ops/deployment/architecture/) for more information).
```
Comment (frankbu) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:12]:
```suggestion
These are then sent to the data plane (see [Architecture](/docs/ops/deployment/architecture/) for more information).
```

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
+---
+title: Configuration Scoping
+description: Shows how to scope configuration in Istio, for operational and performance benefits.
+weight: 10
```
Comment (frankbu) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:4]: If you want this doc to be first in the section then you should make the weight <10, e.g., 5. If you want it last it should be >50. Otherwise you need to pick a number to place it where you want, relative to the other doc weights.
```suggestion
weight: 60
```

File: content/en/docs/ops/deployment/performance-and-scalability/index.md [hunk 6]
```diff
@@ -54,13 +54,13 @@ The CPU consumption scales with the following factors:
...
-a single Istiod instance can support 1000 services, 2000 sidecars with 1 vCPU and 1.5 GB of memory.
 You can increase the number of Istiod instances to reduce the amount of time it takes for the configuration
 to reach all proxies.
 
+At large scale, [configuration scoping](/docs/ops/configuration/mesh) is highly recommended.
```
Comment (craigbox) [inline content/en/docs/ops/deployment/performance-and-scalability/index.md:62]:
```suggestion
At large scale, [configuration scoping](/docs/ops/configuration/mesh/configuration-scoping) is highly recommended.
```

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  egress:
+  - hosts:
+    - "./*" # Import all configuration from our own namespace
+    - "bookinfo/*" # Import all configuration from the bookinfo namespace
+    - "external-services/example.com" # Import only 'example.com' from the external-services namespace
+{{< /text >}}
+
+### `exportTo`
+
+Istio's `VirtualService`, `DestinationRule`, and `ServiceEntry` allow provide a `spec.exportTo` field.
```
Comment (kfaseela) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:54]:
```suggestion
Istio's `VirtualService`, `DestinationRule`, and `ServiceEntry` provide a `spec.exportTo` field.
```

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  Note: changes to `Service` and `Endpoints` will all show as `ServiceEntry`.
+* Istiod exposes metrics `pilot_k8s_cfg_events` and `pilot_k8s_reg_events` for each change.
+* `kubectl get <resource> --watch -oyaml --show-managed-fields` can show changes to an object (or objects) to help understand what is changing, and by whom.
+
+[Headless services](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) (besides ones declared as [HTTP](/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection))
+scale with the number of instances. This makes large headless services expensive, and a good candidate for exclusion with `exportTo` or equivalent.
+
+### What happens if I connect to a service outside of my scope?
+
+When connecting to a service that has been excluded through one of the scoping mechanisms, they data plane will not know anything about the destination,
```
Comment (kfaseela) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:119]:
```suggestion
When connecting to a service that has been excluded through one of the scoping mechanisms, the data plane will not know anything about the destination,
```

File: content/en/docs/ops/configuration/mesh/configuration-scoping/index.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+so it will be treated as [Unmatched traffic](/docs/ops/configuration/traffic-management/traffic-routing/#unmatched-traffic).
+
+### What about Gateways?
+
+While [Gateways](/docs/setup/additional-setup/gateway/) will respect `exportTo` and `DiscoverySelectors`, `Sidecar` objects do not impact Gateways.
+However, unlike sidecars, gateways do not have configuration for the entire cluster by default.
+Instead, each configuration is explicitly attached to the gateway, which mostly avoids this problem.
+
+However, [currently](https://github.com/istio/istio/issues/29131) part of the data plane configuration (a "cluster", in Envoy terms), is always sent for
+the entire cluster, even if its not referenced explicitly.
```
Comment (kfaseela) [inline content/en/docs/ops/configuration/mesh/configuration-scoping/index.md:129]:
```suggestion
the entire cluster, even if it is not referenced explicitly.
```

**Review by dhawton — COMMENTED** [review]
LGTM will let @kfaseela take another look

**Review by kfaseela — APPROVED** [review]
